DataFormats/TotemDigi/interface/TotemTriggerCounters.h

```diff
@@ -0,0 +1,24 @@
+/****************************************************************************
+*
+* This is a part of the TOTEM offline software.
+* Authors: 
+*   Jan Kašpar (jan.kaspar@gmail.com) 
+*    
+****************************************************************************/
+
+#ifndef DataFormats_TotemDigi_TotemTriggerCounters
+#define DataFormats_TotemDigi_TotemTriggerCounters
+
+/**
+ * Trigger counters from LoneG.
+**/
+struct TotemTriggerCounters
+{
+  unsigned char type;
+  unsigned int event_num, bunch_num, src_id;
+  unsigned int orbit_num;
+  unsigned char revision_num;
+  unsigned int run_num, trigger_num, inhibited_triggers_num, input_status_bits;
```

Comment:
On 4/13/16 5:30 AM, jan-kaspar wrote:
> In DataFormats/TotemDigi/interface/TotemTriggerCounters.h
> <https://github.com/cms-sw/cmssw/pull/13837#discussion_r59538756>:
> 
>> +*    
>> +****************************************************************************/
>> +
>> +#ifndef DataFormats_TotemDigi_TotemTriggerCounters
>> +#define DataFormats_TotemDigi_TotemTriggerCounters
>> +
>> +/**
>> + * Trigger counters from LoneG.
>> +**/
>> +struct TotemTriggerCounters
>> +{
>> +  unsigned char type;
>> +  unsigned int event_num, bunch_num, src_id;
>> +  unsigned int orbit_num;
>> +  unsigned char revision_num;
>> +  unsigned int run_num, trigger_num, inhibited_triggers_num, input_status_bits;
> 
> Sorry for my ignorance, but could you please explain me
> 
>   * why struct was OK, while class + public members is not,

struct is strictly speaking "C"
class is "C++"

>   * why encapsulation requires data members to be private,

it's (one of) the definition of encapsulation


>   * why having public setters is any better than having public data members,
>   * why such a simple structure needs to have additional 20 methods with
>     trivial/useless code,
>   * why we can't postpone such "beautification" of the code to a less
>     stressful moment?
> 
> Thanks!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/13837/files/7f46a77fbadfb9757b15e4c9c6b194687f264bc3#r59538756>
>